routers/repo/view.go

```diff
@@ -56,18 +56,31 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 		return
 	}
 
-	var readmeFile *git.Blob
+	// 3 for the extensions in exts[] in order
+	// the last one is for a readme that doesn't
+	// strictly match an extension
+	var readmeFiles [4]*git.Blob
+	var exts = []string{".md", ".txt", ""} // sorted by priority
```

Comment:
Can other file extensions be added: https://stackoverflow.com/a/41112364

Reply:
but it can output the raw content, and if a user has added an external renderer then we don't want them to miss out.

Reply:
@techknowlogick All readme files are detected, it is just about the priority, there a 4th element in `readmeFiles[]` in which any file named `readme` will be detected regardless of its extension.

Reply:
Sounds good.